path_demo.cpp

```diff
@@ -5,44 +5,373 @@
 using namespace std;
 using namespace filesystem;
 
-int main(int argc, char **argv) {
+// TAP-compliant test
```

Comment:
The TAP compliant testing should be its own PR. Keep this PR to the scope of improving CYGWIN on Windows support.

Awesome that you got TAP testing working, though. :)